src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -273,8 +291,11 @@ internal static void CheckForSevereException(Exception e)
             }
             finally
             {
-                TelemetryAPI.ReportExitTelemetry(s_theConsoleHost);
-                s_theConsoleHost.Dispose();
+                if (s_theConsoleHost != null)
```

Comment:
This is the exact condition I am allowing.  The s_theConsoleHost is null because the process is created without a console, but that is Ok if we are running in server mode.  If we are not running in server mode then the HostException is re-thrown since interactive PowerShell requires s_theConsoleHost.

I liked my original changes over this second iteration because it was cleaner.  The command line is parsed and if we run in server mode we don't even bother creating the s_theConsoleHost.  But that version was rejected.

---
In reply to: [85441248](https://github.com/PowerShell/PowerShell/pull/2538#discussion_r85441248) [](ancestors = 85441248)